File: P4/NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'treasure'
require_relative 'monster'
require_relative 'treasure_kind'
require_relative 'combat_result'
require_relative 'dice'
require_relative 'napakalaki_ruby'
require_relative 'player'

module NapakalakiRuby

class CultistPlayer < Player
    attr_accessor :myCultistCard
    
    @@totalCultistPlayers=0
    
  def initialize(p, c)
    super(p.name)
    @myCultistCard=c
    @@totalCultistPlayers=+1
  end
  
  def getCombatLevel
    x=(super.getCombatLevel * 20) / 100
    porcentaje=x.truncate
    
    return super.getCombatLevel + porcentaje + (@myCultistCard.getLevelsGained * @@totalCultistPlayers)
  end
  
  def getOponentLevel(m)
    return m.getLevelChangeAgainstCultistPlayer
  end
  
  def shouldConvert
    return false
  end
  
  def giveMeATreasure
    rnd=Random.new
    tope=super.visibleTreasures.length
    valor=rnd(tope)
    
    return super.visibleTreasures.at(valor)
  end
  
  def canYouGiveMeATreasure
    ser_robado=false
    if super.visibleTreasures.length > 0 then
      ser_robado=true
    end
    return ser_robado
  end
end
end

File: P4/Napakalaki/src/napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.Random;
/**
 *
 * @author pako
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers=0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p); //Herencia de la superclase, llamada al constructor.
        this.myCultistCard=c;
        totalCultistPlayers++;
    }
    
    @Override
    protected int getCombatLevel(){ 
        double x=(super.getCombatLevel() * 20) / 100;
        int porcentaje=(int)x;
        return super.getCombatLevel() + porcentaje + (myCultistCard.getGainedLevels() * totalCultistPlayers);
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    private Treasure giveMeATreasure(){
        Random rnd = new Random();
        int tope=super.visibleTreasures.size();
        int valor= rnd.nextInt(tope);

        return super.visibleTreasures.get(valor);
    }
    
    private boolean canYouGiveMeATreasure(){
        boolean ser_robado=false;
        if(super.visibleTreasures.size()>0)
            ser_robado=true;

        return ser_robado; 
    }
    
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}
